### DISCUSSION TITLE
How improve smoothness of DrawingView?

### DISCUSSION THREAD

**Discussion by Simoo23** [CommunityToolkit/Maui#1756] [Q&A] [answered]
How can I improve the smoothness of the GetImageStream of the DrawingView? 

As you can see the smoothness in the image component is not the same as the DrawingView:
![image](https://github.com/CommunityToolkit/Maui/assets/156714674/d22e12cd-3dbd-4b2c-b032-22ab4783f921)

I set the Source of the image component as follow:
```
var stream = await DrawingView.GetImageStream(Lines, new Size(400, 200), Colors.White);
drawingImage.Source = ImageSource.FromStream(() => stream);
```

In the documentation I found the Granularity property but I didn't understand how to use it.

Have I to implement custom classes for IDrawingLine and IDrawingLineAdapter? 
If yes, can you help me to understand how?

**Comment by VladislavAntonyuk** [comment, marked as answer]
Granularity sets the smooth level of the drawing line.
It adds/removes points to the drawing line.
But the original points collection remains without changes.
As for the image, we draw it as a array of points, and it is not the same as drawing on canvas.
Custom Adapter or DrawingLine are used if you need to adapt your custom Line/Point whatever to our Line. It is a standard Adapter pattern.

**Reply by bit-r** [reply]
This doesn't answer the OP question (which is mine as well), how do you actually use/change the value for the Granularity property used by the DrawingView?
Is Granularity property hardcoded or it can be changed?
If hardcoded, do we have to write our own replacement for IDrawingLine / IDrawingLineAdapter / IDrawingViewHandler?
Any sample code showing the recommended way to do that would be appreciated.

**Reply by VladislavAntonyuk** [reply]
https://github.com/CommunityToolkit/Maui/blob/fbfd456d4ce00bba77a7ec47539dfa39fb6dc682/src/CommunityToolkit.Maui.Core/Views/DrawingView/DrawingLine.shared.cs#L28

**Reply by bit-r** [reply]
Thank you,  I am grateful for the fast reply. 
Will take a look,

**Comment by CliffAgius** [comment]
Closed as answered.
